### ISSUE TITLE
istio-locality label fails k8s validation

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#12582]
https://github.com/istio/istio/blob/70f3993de38ebbdaaee745d9f17a40858abccfae/pilot/pkg/model/service.go#L387-L392

Since LocalityLabel is derived from a K8S labels, which do not allow `/` in them, this isn't very useful. 

```go
	// LocalityLabel indicates the region/zone/subzone of an instance. It is used if the native
	// registry doesn't provide one.
	LocalityLabel = "istio-locality"
```

We can't actually define `region/zone/subzone` for an instance, because it will fail validation. 

It may also be useful to let this override K8S definition, if it is provided.

@liamawhite

**Comment by liamawhite** [comment]
Maybe we should change to `~`?

**Comment by howardjohn** [comment]
Seems reasonable to me. I think the ideal solution would be an annotation, but the current code only grabs all of the labels - not sure how much effort adding annotations would be but I don't think it is that straight forward.

**Comment by hzxuzhonghu** [comment]
`~`  is also not a valid character. ` regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')`

So the only choice we should use is `_`  or `.`

**Comment by howardjohn** [comment]
Fixed in 1.1.1
